### ISSUE TITLE
Potential indexing error in managelayers.

### ISSUE THREAD

**Issue by chadagreene** [alex-s-gardner/GEMB#1]
https://github.com/alex-s-gardner/GEMB/blob/7b05ea5bf9dcb03cdb45c8b3fc26a5a54f1e3a39/GEMB/managelayers.m#L105C21-L105C27

@alex-s-gardner I am confused by Line 105 of `managelayers` and I wonder if it's a mistake. It says: 
```matlab
dzMax2(i)=max(zY2*dzMin2(i-1),dzMin*2.0);
```
The index referencing `dzMin2` is surprising because `dzMin2` can have totally different dimensions from the `dzMax2` array. That's because a few lines before this loop, a bunch of cells were combined and array elements were deleted with: 
```matlab 
% delete combined cells
D = (m <= Delflag+Wtol);
m(D) = []; W(D) = []; dz(D) = []; d(D) = []; T(D) = []; a(D) = [];
re(D) = []; gdn(D) = []; gsp(D) = []; adiff(D)=[]; EI(D)=[]; EW(D)=[];
```
In Line 105, `dzMin2` is the size of the arrays before cells were deleted, whereas `dzMax2` is the size of arrays after cells were deleted. 

I am wondering if Line 105 is supposed to reference `dzMax2` instead of `dzMin2`? Or if the same elements of `dzMin2` should be deleted on line 89 via:
```matlab
dzMin2(D) = [];
```

Can you take a look and let me know what you think?

**Comment by alex-s-gardner** [comment]
@chadagreene in all honesty I'm a bit turned around looking at the code. It seems strange that on line [110](https://github.com/alex-s-gardner/GEMB/blob/7b05ea5bf9dcb03cdb45c8b3fc26a5a54f1e3a39/GEMB/managelayers.m#L110) that dzMax2 is again multiplied by `zY2`. 

Why didn't I just write line 105 as:
`dzMax2(i)=max(zY2*dzMin2(i), dzMin*2.0);` 

and then line 110 as:
`if dz(j) > dzMax2(j)+Dtol`

Regardless, I think it's a safe to assume that the same elements of dzMin2 should be deleted on line 89

**Comment by chadagreene** [comment]
@alex-s-gardner The `i-1` indexing on line 105 follows a similar form that was used on line [44](https://github.com/alex-s-gardner/GEMB/blob/7b05ea5bf9dcb03cdb45c8b3fc26a5a54f1e3a39/GEMB/managelayers.m#L44): 
```matlab
dzMin2(i)=zY2*dzMin2(i-1);
```
Does that bring any insightful context?

**Comment by alex-s-gardner** [comment]
@chadagreene Yes I understand:

`dzMin2` defines the minimum allowable cell size:
 `dzMin2(i) = zY2*dzMin2(i-1)`.. this simply grows the minimum allowable cell size linearly with depth, starting below the equally spaced "surface layer"
 
`dzMax2` defines the maximum allowable cell size:
`dzMax2(i)=max(zY2*dzMin2(i-1),dzMin*2.0)` which is equivalent to `max(dzMin2(i),dzMin*2.0)` note change in indexing 

This means that below the "surface layer" `dzMax2(i)` == `dzMin2(i)` which is not correct and why there is a extra `zY2` in line [110](https://github.com/alex-s-gardner/GEMB/blob/7b05ea5bf9dcb03cdb45c8b3fc26a5a54f1e3a39/GEMB/managelayers.m#L110)

A more appropriate way would have been to define `dzMax2` correctly in the first place:
`dzMax2(i)=max(zY2*dzMin2(i), dzMin*2.0)`
 and to drop the extra `zY2` on line 110
 
 

But this is all in addition to what you noticed: **that the same elements of dzMin2 should be deleted on line 89**

**Comment by chadagreene** [comment]
@alex-s-gardner I've made a few edits to `managelayers`. Most of the edits have been for efficiency and readability, and have not impacted the output of the function. However, the new edit on line [99](https://github.com/alex-s-gardner/GEMB/blob/29eb9cdfa41fe6d7c2b795a6dcd2bf37589a3925/GEMB/managelayers.m#L99) _does_ impact the output of `managelayers` and the `GEMB` function. 

Can you confirm that the change I made on line 99 is correct or if I should revert it?

**Comment by alex-s-gardner** [comment]
@chadagreene this change is correct

**Comment by chadagreene** [comment]
The original issue was addressed in the Matlab version in commit 48f01ee. The same issue will be addressed in the C++ version in Issue #5.
